apps/frontend/src/pages/auth/libs/components/sign-in-form/sign-in-form.tsx

```diff
@@ -1,29 +1,41 @@
-import { Button, Input } from "~/libs/components/components.js";
-import { useAppForm, useCallback, useState } from "~/libs/hooks/hooks.js";
+import { Button, Input, Link } from "~/libs/components/components.js";
+import { AppRoute } from "~/libs/enums/app-route.enum.js";
+import {
+	useAppDispatch,
+	useAppForm,
+	useCallback,
+	useState,
+} from "~/libs/hooks/hooks.js";
+import { actions as authActions } from "~/modules/auth/auth.js";
 import {
 	type UserSignInRequestDto,
 	userSignInValidationSchema,
 } from "~/modules/users/users.js";
 
-import { DEFAULT_SIGN_IN_PAYLOAD } from "./libs/constants.js";
+import { DEFAULT_SIGN_IN_PAYLOAD } from "./libs/constants/constants.js";
 import styles from "./styles.module.css";
 
-type Properties = {
-	onSubmit: (payload: UserSignInRequestDto) => void;
-};
-
-const SignInForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
+const SignInForm: React.FC = () => {
 	const { control, errors, handleSubmit } = useAppForm<UserSignInRequestDto>({
 		defaultValues: DEFAULT_SIGN_IN_PAYLOAD,
 		validationSchema: userSignInValidationSchema,
 	});
 	const [isPasswordVisible, setIsPasswordVisible] = useState<boolean>(false);
 
+	const dispatch = useAppDispatch();
+
+	const handleSignInSubmit = useCallback(
```

Comment:
Why did you move it here?

Reply:
You mean hooks or submit handlers? I moved submit handlers because I felt like it was a logical thing to do. These handlers don't really have to do anything with auth component, and hanlder for reset passwords requires parameters that are accessible only on reset password form submit handler. It felt weird to move one handler into component and leave others in the parent component

Reply:
Let's keep everything in auth component. For reset password, do the same as for sign in/sign up - send the required parameters as arguments to a function

Reply:
I will do that, but I want to ask why is this necessary? Having 4 function definitions in the file convolutes the code in my opinion

Reply:
We want to keep functions in auth component, because the main goal of other component is to render the screen, not have logic. In some places we will even keep the redux state in one parent file and send as props to children